packages/client/src/features/viewport/viewport-tool.ts

```diff
@@ -0,0 +1,50 @@
+/********************************************************************************
+ * Copyright (c) 2025 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { TYPES } from '@eclipse-glsp/sprotty';
+import { inject, injectable } from 'inversify';
+import type { IShortcutManager } from '../shortcuts/shortcuts-manager';
+import { BaseTool } from '../tools/base-tools';
+import { MoveViewportKeyListener, ZoomKeyListener } from './viewport-key-listener';
+
+@injectable()
+export class ViewportTool extends BaseTool {
```

Comment:
Maybe rename it to `ViewportKeyTool` and add  short doc comment to mention that this tool only handles keyboard related viewport actions. Mouse actions for zooming/paning are currently bound as direct listeners and therefore alway on.